src/messaging.rs

```diff
@@ -25,6 +26,32 @@ pub enum Target<N> {
     Node(N),
 }
 
+impl<N: Ord> Ord for Target<N> {
+    fn cmp(&self, other: &Target<N>) -> Ordering {
+        match (self, other) {
+            (Target::Node(id1), Target::Node(id2)) => id1.cmp(id2),
+            (Target::All, Target::All) => Ordering::Equal,
+            (Target::All, Target::Node(_)) => Ordering::Greater,
+            (Target::Node(_), Target::All) => Ordering::Less,
+        }
+    }
+}
+
+impl<N: PartialEq> PartialOrd for Target<N> {
+    fn partial_cmp(&self, other: &Target<N>) -> Option<Ordering> {
+        match (self, other) {
+            (Target::Node(id1), Target::Node(id2)) => if id1 == id2 {
+                Some(Ordering::Equal)
+            } else {
+                None
+            },
+            (Target::All, Target::All) => Some(Ordering::Equal),
+            (Target::All, Target::Node(_)) => Some(Ordering::Greater),
+            (Target::Node(_), Target::All) => Some(Ordering::Less),
+        }
```

Comment:
That should probably just be `Some(self.cmp(other))`, or rather: be derived?

Reply:
Same as above. Can be removed.